### DISCUSSION TITLE
Consultar conteúdo de uma NF

### DISCUSSION THREAD

**Discussion by grammobrasil** [TadaSoftware/PyNFe#179] [Q&A] [answered]
Bom dia. É possível consultar o conteúdo ("produtos", valores, etc) de uma NF de Saída emitidas por terceiros dentro do "Consulta Nota"?

**Comment by leogregianin** [comment]
> Bom dia. É possível consultar o conteúdo ("produtos", valores, etc) de uma NF de Saída emitidas por terceiros dentro do "Consulta Nota"?

@grammobrasil as regras definidas pela Receita Federal/SEFAZ nas Notas Técnicas é não permitir consultar notas de terceiros, você somente pode consultar as notas pertencentes ao seu certificado digital. O serviço `Consulta Nota` funciona dessa forma.

Inclusive, a partir da entrada em vigor da [LGPD](http://www.planalto.gov.br/ccivil_03/_ato2015-2018/2018/lei/l13709.htm), as Notas Fiscais Eletrônicas somente poderão ser acessados pelas partes envolvidas, ou seja, emitente, destinatário, transportador e demais autorizados através do serviço [DistribuicaoDFe](https://github.com/TadaSoftware/PyNFe/wiki/Consulta-Distribui%C3%A7%C3%A3o).

**Reply by grammobrasil** [reply]
Certo, mas justamente gostaria de consultar uma nota emitida para nossa empresa, porém os detalhes dela.
Em especial desejo integrar dados das compras de produtos ao sistema de estoque. 
Não encontrei como verificar essa informação no Consulta Nota.
Peço desculpas se estou fazendo algo muito equivocado, estamos aprendendo ainda bastante na área, mas agradeço qualquer ajuda.

**Reply by leogregianin** [reply]
> 

O serviço `Consulta Nota` é para consultar notas emitidas `pelo` seu próprio CNPJ.

O serviço para consultar notas emitidas `para` o seu CNPJ é o DistribuicaoDFe, que é um serviço do ambiente nacional que verifica todas as notas emitidas do Brasil inteiro `para` o seu CNPJ.

Nesse [link](https://github.com/TadaSoftware/PyNFe/wiki/Fluxo-da-Consulta-de-Distribuicao) tem um exemplo de fluxo de como consultar e baixar o XML completo da nota.

Porém, antes que o serviço permita você baixar a Nota completa, é necessário realizar o envio do evento de manifestação. Nesse [link](https://github.com/TadaSoftware/PyNFe/wiki/Manifesta%C3%A7%C3%A3o-Destinat%C3%A1rio) tem um exemplo de como realizar isso. 

Mas antes de tudo, recomendo a leitura das Notas Técnicas ([nesse link](https://www.nfe.fazenda.gov.br/portal/listaConteudo.aspx?tipoConteudo=04BIflQt1aY=)) referente a DistribuiçãoDFe, como funciona, como se utiliza, suas limitações, etc.

**Comment by grammobrasil** [comment, marked as answer]
Muito obrigado e desculpa a confusão.
Eu realmente estava confundindo os conceitos.
Dei uma estudada melhor e agora estou conseguindo utilizar adequadamente o script.
Abraço.

**Reply by leogregianin** [reply]
@grammobrasil, para melhor organização das discussões do projeto, pode marcar essa discussão por respondida?
